lib/iris/_concatenate.py

```diff
@@ -917,97 +1062,79 @@ def register(
             elif not match:
                 mismatch_error_msg = f"Found cubes with overlap on concatenate axis {candidate_axis}, skipping concatenation for these cubes"
 
-        # Check for compatible AuxCoords.
-        if match:
-            if check_aux_coords:
-                for coord_a, coord_b in zip(
-                    self._cube_signature.aux_coords_and_dims,
-                    cube_signature.aux_coords_and_dims,
+        def get_hash(array):
+            return hashes[array_id(array)]
+
+        def get_hashes(coord):
+            result = []
+            if hasattr(coord, "core_points"):
+                result.append(get_hash(coord.core_points()))
+                if coord.has_bounds():
+                    result.append(get_hash(coord.core_bounds()))
+            else:
+                result.append(get_hash(coord.core_data()))
+            return tuple(result)
+
+        # Mapping from `_CubeSignature` attributes to human readable names.
+        coord_type_names = {
+            "aux_coords_and_dims": "Auxiliary coordinates",
+            "cell_measures_and_dims": "Cell measures",
+            "ancillary_variables_and_dims": "Ancillary variables",
+            "derived_coords_and_dims": "Derived coordinates",
+        }
+
+        def check_coord_match(coord_type):
+            for coord_a, coord_b in zip(
+                getattr(self._cube_signature, coord_type),
+                getattr(cube_signature, coord_type),
+            ):
+                # Coordinates that span the candidate axis can differ
+                if (
+                    candidate_axis not in coord_a.dims
+                    or candidate_axis not in coord_b.dims
                 ):
-                    # AuxCoords that span the candidate axis can differ
-                    if (
-                        candidate_axis not in coord_a.dims
-                        or candidate_axis not in coord_b.dims
+                    if not (
+                        coord_a.dims == coord_b.dims
+                        and get_hashes(coord_a.coord) == get_hashes(coord_b.coord)
```

Comment:
I've looked at how `_cube_signature` is defined, and I think this might be a change in scope. In the past, comparing `coord_a == coord_b` would have invoked `DimensionalMetadata.__eq__()`, which includes checking the `metadata` attribute. Now we are only comparing the array content.

But I have some doubts, since making such a change would have presumably caused test failures. Can you persuade me that I am wrong?

https://github.com/SciTools/iris/blob/5c18504eaecfd92651c3e408ce14c1cba9caf47a/lib/iris/coords.py#L578-L608

Reply:
This code already compares the coordinate metadata (and `dims`, so I can take that out too):
https://github.com/SciTools/iris/blob/b8f554f7b8899f2ff6754857d59fc48944ef8e9b/lib/iris/_concatenate.py#L893-L895
as you can see in `_CubeSignature.match` here:
https://github.com/SciTools/iris/blob/b8f554f7b8899f2ff6754857d59fc48944ef8e9b/lib/iris/_concatenate.py#L564-L589
and to understand that it helps to see how metadata is getting attached to the `_CubeSignature` in its `__init__` method:
https://github.com/SciTools/iris/blob/b8f554f7b8899f2ff6754857d59fc48944ef8e9b/lib/iris/_concatenate.py#L440-L444
with the actual metadata etc comparison getting called from `_CoordMetaData.__eq__`:
https://github.com/SciTools/iris/blob/b8f554f7b8899f2ff6754857d59fc48944ef8e9b/lib/iris/_concatenate.py#L120-L126

Reply:
Great! Thanks for the clarification. I agree with your explanation.